daal4py/sklearn/neighbors/knn_classifier.py

```diff
@@ -0,0 +1,546 @@
+                    "enter integer value" %
+                    type(n_neighbors))
+
+        if X is not None:
+            query_is_train = False
+            if self.effective_metric_ == 'precomputed':
+                X = _check_precomputed(X)
+            else:
+                X = check_array(X, accept_sparse='csr')
+        else:
+            query_is_train = True
+            X = self._fit_X
+            # Include an extra neighbor to account for the sample itself being
+            # returned, which is removed later
+            n_neighbors += 1
+
+        n_samples_fit = self.n_samples_fit_
+        if n_neighbors > n_samples_fit:
+            raise ValueError(
+                "Expected n_neighbors <= n_samples, "
+                " but n_samples = %d, n_neighbors = %d" %
+                (n_samples_fit, n_neighbors)
+            )
+
+        n_jobs = effective_n_jobs(self.n_jobs)
+        chunked_results = None
+
+        try:
+            fptype = getFPType(X)
+        except ValueError:
+            fptype = None
+
+        if self._fit_method in ['brute'] \
+        and (self.effective_metric_ == 'minkowski' and self.p == 2 or self.effective_metric_ == 'euclidean') \
+        and fptype is not None:
+
+            if self._fit_method == 'brute':
+                knn_classification_training = d4p.bf_knn_classification_training
+                knn_classification_prediction = d4p.bf_knn_classification_prediction
+            else:
+                knn_classification_training = d4p.kdtree_knn_classification_training
+                knn_classification_prediction = d4p.kdtree_knn_classification_prediction
+
+            training_alg = knn_classification_training(
+                fptype=fptype,
+                method='defaultDense'
+            )
+
+            fit_X = d4p.get_data(self._fit_X)
+            training_result = training_alg.compute(fit_X, np.empty(shape=(fit_X.shape[0], 1)))
+
+            prediction_alg = knn_classification_prediction(
+                fptype=fptype,
+                method='defaultDense',
+                k=n_neighbors,
+                # resultsToCompute='computeIndicesOfNeightbors|computeDistances'
+            )
+
+            X = d4p.get_data(X)
+            prediction_result = prediction_alg.compute(X, training_result.model)
+
+            # with open("/data/datae/onedal_python_benchmarks/knn_tests/bf/knn_log.txt", "a") as log:
+            #     log.write("unsup,{},{},{},{},{},{},{},{}\n".format(self._fit_method, self.effective_metric_, fptype, n_neighbors,
+            #                                                        str(type(prediction_result)), str(type(training_result.model)),
+            #                                                        str(knn_classification_training), str(knn_classification_prediction)))
+
+            if return_distance:
+                results = prediction_result.distances, prediction_result.indices
```

Comment:
Are these tables returned by default? Are you sure?

Reply:
Yes, same as stock implementation